ocp_resources/cluster_role.py

```diff
@@ -33,19 +33,46 @@ def __init__(
         self.api_groups = api_groups
         self.permissions_to_resources = permissions_to_resources
         self.verbs = verbs
+        self.desired_state = {"rules": []}
 
     def to_dict(self):
-        res = super().to_dict()
+        self.res = super().to_dict()
         if self.yaml_file:
-            return res
+            return self.res
 
-        rules = {}
-        if self.api_groups:
-            rules["apiGroups"] = self.api_groups
         if self.permissions_to_resources:
-            rules["resources"] = self.permissions_to_resources
-        if self.verbs:
-            rules["verbs"] = self.verbs
-        if rules:
-            res["rules"] = [rules]
-        return res
+            self.add_rule(
+                api_groups=self.api_groups,
+                permissions_to_resources=self.permissions_to_resources,
+                verbs=self.verbs,
+            )
+
+        return self.res
+
+    def add_rule(
+        self,
+        api_groups=None,
+        permissions_to_resources=None,
+        verbs=None,
+    ):
+        if not self.res:
+            self.res = self.to_dict()
+
+        rule = {}
+        if api_groups:
+            rule["apiGroups"] = api_groups
+        if permissions_to_resources:
+            rule["resources"] = permissions_to_resources
+        if verbs:
+            rule["verbs"] = verbs
+        if rule:
+            self.set_rule(rule=rule)
+
+        return self.res
+
+    def set_rule(self, rule):
```

Comment:
Is there any point to keep this public? User can call set_rule()? 
I think it should be private, and then you can remove 
```
if not self.res:
            self.res = super().to_dict()
```

Reply:
+1